### PR TITLE
SAMZA-1821: fix the IllegalStateException complaining duplicate key when fetching systemStream configs

### PR DESCRIPTION
**Pull request by weiqingy**
## What changes were proposed in this pull request?
This PR is to fix IllegalStateException complaining duplicate key when fetching resource configs in SamzaSqlApplicationConfig.

## How was this patch tested?
Pass the local build and current unit tests.
Added a new unit test.

### CODE REVIEW

File: samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationConfig.java [hunk 1]
```diff
@@ -113,13 +113,13 @@ public SamzaSqlApplicationConfig(Config staticConfig) {
     inputSystemStreamConfigBySource = queryInfo.stream()
         .map(QueryInfo::getSources)
         .flatMap(Collection::stream)
-        .collect(Collectors.toMap(Function.identity(), src -> ioResolver.fetchSourceInfo(src)));
+        .collect(Collectors.toMap(Function.identity(), ioResolver::fetchSourceInfo, (src1, src2) -> src1));
```
Comment (bharathkk) [inline samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationConfig.java]: How about we use `distinct()` before `collect()` to save some resolver calls to fetch source information?

File: samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationConfig.java [hunk 2]
```diff
@@ -113,13 +113,13 @@ public SamzaSqlApplicationConfig(Config staticConfig) {
...
 
     Set<SqlIOConfig> systemStreamConfigs = new HashSet<>(inputSystemStreamConfigBySource.values());
 
     outputSystemStreamConfigsBySource = queryInfo.stream()
         .map(QueryInfo::getSink)
-        .collect(Collectors.toMap(Function.identity(), x -> ioResolver.fetchSinkInfo(x)));
+        .collect(Collectors.toMap(Function.identity(), ioResolver::fetchSinkInfo, (src1, src2) -> src1));
```
Comment (bharathkk) [inline samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationConfig.java]: same comment as above.

File: samza-sql/src/test/java/org/apache/samza/sql/testutil/TestSamzaSqlFileParser.java [hunk 3]
```diff
@@ -24,7 +24,6 @@
 import java.io.PrintWriter;
 import java.util.List;
 
-import org.apache.samza.sql.testutil.SqlFileParser;
```
Comment (bharathkk) [inline samza-sql/src/test/java/org/apache/samza/sql/testutil/TestSamzaSqlFileParser.java]: nit: revert

File: samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationConfig.java (hunk 1 above)
Comment (weiqingy) [reply samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationConfig.java]: right. thanks for pointing this out.

File: samza-sql/src/test/java/org/apache/samza/sql/testutil/TestSamzaSqlFileParser.java (hunk 3 above)
Comment (weiqingy) [reply samza-sql/src/test/java/org/apache/samza/sql/testutil/TestSamzaSqlFileParser.java]: Just cleaning up code when I see it.

**Pull request closed without merging**